File: frontend/src/app/interfaces/appointment-response.ts
import { AppointmentStatus } from "./appointment-status";

export interface AppointmentResponse {
    
    id: number;

    //  Appointment details

    startDateTime: string;
    durationInSeconds: number;
    appointmentStatus: AppointmentStatus;

    // Patient details

    patientFullName: string;
    patientEmail: string;
    patientPhoneNumber: string;
    patientReason: string;
    clinicianNotes: string;

    // Audit details

    isAcknowledged: boolean;
    createdAt: string;
    updatedAt: string;
    
}

File: backend/src/main/java/com/hear2speak/dtos/AppointmentResponse.java
package com.hear2speak.dtos;

import com.hear2speak.entities.AppointmentStatus;

public class AppointmentResponse {

    public Long id;

    // Appointment details
    
    public String startDateTime;

    public Integer durationInSeconds;

    public AppointmentStatus appointmentStatus;

    // Patient details

    public String patientFullName;

    public String patientEmail;

    public String patientPhoneNumber;

    public String patientReason;

    public String clinicianNotes;

    // Audit details

    public Boolean isAcknowledged;

    public String createdAt;

    public String updatedAt;

}
